py3status/modules/arch_updates.py

```diff
@@ -50,83 +39,88 @@ class Py3status:
     cache_timeout = 600
     format = None
     hide_if_zero = False
-    include_aur = False
+
+    class Meta:
+        deprecated = {"remove": [{"param": "include_aur", "msg": "obsolete"}]}
 
     def post_config_hook(self):
-        if not self.format:
-            if not self.include_aur:
-                self.format = FORMAT_PACMAN_ONLY
-            else:
-                self.format = FORMAT_PACMAN_AND_AUR
-        self.include_aur = self.py3.format_contains(self.format, "aur")
-        self.include_pacman = self.py3.format_contains(self.format, "pacman")
-        if self.py3.format_contains(self.format, "total"):
-            self.include_aur = True
-            self.include_pacman = True
-
-        if self.include_aur:
-            aurs = ["auracle", "yay", "cower"]
-            command = self.py3.check_commands(aurs)
-            if command:
-                self._check_aur_updates = getattr(
-                    self, "_check_aur_updates_{}".format(command)
-                )
+        helper = {
+            "pacman": self.py3.check_commands(["checkupdates"]),
+            "aur": self.py3.check_commands(["auracle", "yay", "cower"]),
+        }
+        if self.format:
+            placeholders = self.py3.get_placeholders_list(self.format)
+            if "total" in placeholders:
+                pass
+            elif not any(helper.values()):
+                raise Exception(STRING_NOT_INSTALLED.format("pacman, aur"))
             else:
-                self.include_aur = False
-                self.py3.notify_user(STRING_NOT_INSTALLED.format(", ".join(aurs)))
-
-    def arch_updates(self):
-        pacman_updates = aur_updates = total = None
-
-        if self.include_pacman:
-            pacman_updates = self._check_pacman_updates()
+                for name in helper:
+                    if name not in placeholders:
+                        helper[name] = None
+                    elif not helper[name]:
+                        raise Exception(STRING_NOT_INSTALLED.format(name))
+        elif all(helper.values()):
+            self.format = "UPD: {pacman}/{aur}"
+        elif helper["pacman"]:
+            self.format = "UPD: {pacman}"
+        elif helper["aur"]:
+            self.format = "UPD: ?/{aur}"
+        else:
+            raise Exception(STRING_NOT_INSTALLED.format("pacman, aur"))
 
-        if self.include_aur:
-            aur_updates = self._check_aur_updates()
+        m = "_get_{}_updates"
+        self._get_pacman_updates = getattr(self, m.format(helper["pacman"]))
+        self._get_aur_updates = getattr(self, m.format(helper["aur"]))
 
-        if pacman_updates is not None or aur_updates is not None:
-            total = (pacman_updates or 0) + (aur_updates or 0)
+    def _get_None_updates(self):
+        pass
```

Comment:
this looks like an ugly hack, can we instead check for None before trying to execute some reflection magic?

Reply:
Sure. :)